common.c

```diff
@@ -85,66 +178,17 @@ trace_printf(int hdr, const char *fmt, ...)
 	real_getpid = RETRACE_GET_REAL(getpid);
 
 	va_start(arglist, fmt);
-
-	memset(str, 0, sizeof(str));
-
 	real_vsnprintf(str, sizeof(str), fmt, arglist);
 
-	str[sizeof(str) - 1] = '\0';
-
 	if (hdr == 1)
 		real_fprintf(stderr, "(%d) ", real_getpid());
 
-	real_fprintf(stderr, "%s", str);
-
+	trace_printf_str(str);
 	va_end(arglist);
 
 	set_tracing_enabled(old_tracing_enabled);
-}
-
-void
-trace_printf_str(const char *string)
-{
-	static const char CR[] = VAR "\\r" RST;
-	static const char LF[] = VAR "\\n" RST;
-	static const char TAB[] = VAR "\\t" RST;
-	static const char SNIP[] = "[SNIP]";
```

Comment:
The whole point of this function was the [SNIP] stuff, which you didn't carry over to your rewrite.